src/envoy/http/metadata_exchange/plugin.cc

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NULL_PLUGIN
+#include "plugin.h"
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+
+// clang-format off
+constexpr char CHAR_TABLE[] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+constexpr unsigned char REVERSE_LOOKUP_TABLE[256] = {
```

Comment:
I think this can go into a separate header / cc file.
It clutters the main file too much .

Reply:
Agree.